index.js

```diff
@@ -74,12 +74,37 @@ const setupPermissions = async (repoName, pullAccountIds) => {
     ]
   };
 
-  core.info(JSON.stringify(policy));
-  const params = {
-    "repositoryName": repoName,
-    "policyText": JSON.stringify(policy)
-  };
-  await ecr.send(new SetRepositoryPolicyCommand(params));
+  let newPolicyText = JSON.stringify(policy);
+  let updateLifecyclePolicy = false;
+
+  // Do we need to update LifeCycle Policy?
+  try {
+    r = await ecr.getRepositoryPolicy({ repositoryName: repoName })
+    let oldPolicyText = JSON.stringify(JSON.parse(r.policyText))
+
+    // Update only if Policy changed
+    if (oldPolicyText !== newPolicyText) {
```

Comment:
JSON.stringify(JSON.parse())? is r.policyText === oldPolicyText?

are u sure that it will be stable in terms of order of fields? looks tricky.

const person1 = {
    "firstName": "John",
    "lastName": "Doe",
    "age": 35 
}

const person2 = {
    "lastName": "Doe",
    "firstName": "John",
    "age": 35 
}

JSON.stringify(person1) === JSON.stringify(person2); // false